### DISCUSSION TITLE
tuning hyperparameters and integration with other libraries

### DISCUSSION THREAD

**Discussion by georgemossad** [timeseriesAI/tsai#669] [Q&A] [answered]
hey there hope you are doing good,
I’m using tsai and I like it very much but I wonder if I can tune the hyperparameters in the models and also if i can use the model architectures in other libraries like keras.
thanks

**Comment by oguiza** [comment, marked as answer]
Hi @georgemossad,
You may want to take a look at a [tutorial notebook on hyperparameter optimization](https://github.com/timeseriesAI/tsai/blob/45fb7c8b2da6ba47bd625e10c600083333507240/tutorial_nbs/13_Hyperparameter_optimization_with_Optuna.ipynb). There we demonstrate how you can use Optuna to tune your hyperparameters. 
As to the question on keras, I'm not sure I understand. If you mean if you can use any Keras model with tsai, the answer is no. You can only use Pytorch models.

**Reply by georgemossad** [reply]
that's great thanks, no problem i think I'm more comfortable now to tsai than keras.
thanks for this amazing work,i just wonder after training the model and save it can i use it in other frameworks

**Reply by oguiza** [reply]
Once you train the model, you'll be able to convert it to torchscript that is independent of the framework.
